### DISCUSSION TITLE
Removing a Node does not work

### DISCUSSION THREAD

**Discussion by LookingForAnswers** [Security-Onion-Solutions/securityonion#6017] [Unsupported Versions] [answered]
Hello,

I have followed the instructions here [https://docs.securityonion.net/en/2.3/removing-a-node.html#removing-a-node](https://docs.securityonion.net/en/2.3/removing-a-node.html#removing-a-node) but the sensor node continues to show up in Grafana, Grid, and FleetDM.

I have replaced the old sensor with a new sensor install and hardware. The new sensor works without issue.

Thank you in advance for your help,
John

**Comment by jertel** [comment]
If the old sensor has been shutdown then running `so-soc-restart` will clear it out of Grid. Others may be able to offer insight into Grafana and FleetDM.

**Reply by LookingForAnswers** [reply]
Thanks Jason,

The `so-soc-restart` solved the Grid problem.

**Comment by LookingForAnswers** [comment]
I have restarted the manager previously but maybe the sensor was still on during the reboot.

I can remove the forward node through the FleetDM web ui but I was assuming following the instructions would remove the node everywhere.

I'm hoping others will have the answer as I expect replacing or moving sensors will be something I will need to do again.

**Comment by LookingForAnswers** [comment, marked as answer]
For other's reference, I was able to remove the old sensor from Grid, Grafana, and Fleet

- Grafana: I purged the data using `so-influxdb-clean` described here [https://docs.securityonion.net/en/2.3/grafana.html](https://docs.securityonion.net/en/2.3/grafana.html) to a time after the old sensor was turned off
- Fleet: used the FleetDM web ui and deleted the node
- Grid: The `so-soc-restart` solved the Grid problem once the sensor was turned off. (thanks Jason)

I was not able to remove the agent from Wazuh  [https://docs.securityonion.net/en/2.3/wazuh.html?highlight=wazuh#adding-agents](https://docs.securityonion.net/en/2.3/wazuh.html?highlight=wazuh#adding-agents) as described here:
"Please keep in mind that when you run manage_agents you will need to do so inside the so-wazuh container like this: `sudo so-wazuh-agent-manage` "

The `sudo so-wazuh-agent-manage` command does not show all the agents only the so-manager but I'm not familiar with docker and maybe I'm not running the command in the right place. Also removing the Wazuh agent from the manager may not be needed but just wanted to clean up all the references.

**Reply by dougburks** [reply]
> For other's reference, I was able to remove the old sensor from Grid, Grafana, and Fleet

I've updated the documentation:
https://docs.securityonion.net/en/2.3/removing-a-node.html

> The `sudo so-wazuh-agent-manage` command does not show all the agents only the so-manager but I'm not familiar with docker and maybe I'm not running the command in the right place. Also removing the Wazuh agent from the manager may not be needed but just wanted to clean up all the references.

Each node has its own internal Wazuh server, so your old node was never an agent of the Wazuh server on your manager node.
